packages/flutterfire_cli/lib/src/firebase/firebase_apple_setup.dart

```diff
@@ -0,0 +1,654 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:cli_util/cli_logging.dart';
+import 'package:path/path.dart' as path;
+
+import '../common/strings.dart';
+import '../common/utils.dart';
+import '../firebase/firebase_options.dart';
+
+import '../flutter_app.dart';
+
+// Use for both macOS & iOS
+class FirebaseAppleSetup {
+  FirebaseAppleSetup(
+    this.platformOptions,
+    this.flutterApp,
+    this.fullPathToServiceFile,
+    this.googleServicePathSpecified,
+    this.logger,
+    this.generateDebugSymbolScript,
+    this.buildConfiguration,
+    this.target,
+    this.platform,
+  );
+  // Either "iOS" or "macOS"
+  final String platform;
+  final FlutterApp? flutterApp;
+  final FirebaseOptions platformOptions;
+  String? fullPathToServiceFile;
+  bool googleServicePathSpecified;
+  final Logger logger;
+  final bool? generateDebugSymbolScript;
+// This allows us to update to the required "GoogleService-Info.plist" file name for iOS target or build configuration writes.
+  String? updatedServiceFilePath;
+  String? buildConfiguration;
+  String? target;
+
+  String get xcodeProjFilePath {
```

Comment:
There seems to be a lot of internal methods in this file that pass around `String xcodeProjFilePath` as an arg, could they not be removed as an arg and just use this getter?